File: Problems/Algorithms/705. Design HashSet/DesignHashSet.kt
class MyHashSet() {
    
    private val HASHSIZE = 256
    private val hashSet: Array<MutableList<Int>> = Array(HASHSIZE) { mutableListOf<Int>() }
    private var curr: MutableList<Int> = mutableListOf()

    fun add(key: Int) {
        if (!contains(key)) {
            curr.add(key)
        }
    }

    fun remove(key: Int) {
        if (contains(key)) {
            curr.remove(key)
        }
    }

    fun contains(key: Int): Boolean {
        curr = hashSet[key % HASHSIZE]
        return curr.contains(key)
    }

}

/**
 * Your MyHashSet object will be instantiated and called as such:
 * var obj = MyHashSet()
 * obj.add(key)
 * obj.remove(key)
 * var param_3 = obj.contains(key)
 */

File: Problems/Algorithms/705. Design HashSet/design_hashset.py
class MyHashSet:

    def __init__(self):
        self.hs = 0

    def add(self, key: int) -> None:
        self.hs |= (1 << key)

    def remove(self, key: int) -> None:
        self.hs &= ~(1 << key)

    def contains(self, key: int) -> bool:
        return (self.hs >> key) & 1


# Your MyHashSet object will be instantiated and called as such:
# obj = MyHashSet()
# obj.add(key)
# obj.remove(key)
# param_3 = obj.contains(key)
